schematic/store/synapse.py

```diff
@@ -1430,13 +1430,15 @@ def buildDB(
     @tracer.start_as_current_span("SynapseStorage::upload_manifest_file")
     def upload_manifest_file(
         self,
-        manifest,
-        metadataManifestPath,
+        manifest: pd.DataFrame,
+        metadataManifestPath: str,
         datasetId,
         restrict_manifest,
         component_name="",
     ):
         # Update manifest to have the new entityId column
+        # TODO: Update this code to allow saving the manifest to disk via an in memory buffer
+        # so that we can avoid writing to disk
         manifest.to_csv(metadataManifestPath, index=False)
```

Comment:
This is optional (Feel free to remove the TODO). It's something that would need to be implemented within synpy and is quite a large change to make this happen.